addons/obiaway/Obiaway.lua

```diff
@@ -27,109 +27,136 @@
--- to day/weather/storm effect change. Uses itemizer.
--- Intended to be used in conjunection with spellcast xml that
--- moves needed obis from sack to inventory just in time. 
+-- Puts elemental obi away when they are no longer needed due
+-- to day/weather/storm effect change and gets elemental obi based
+-- on same conditions. Uses itemizer addon.
 -- 
--- The advantage of this system compared to moving obis back during 
+-- The advantage of this system compared to moving obi back during 
 -- aftercast is that it avoids excessive inventory movement,
 -- so malfunctions due to inventory filling up completely are
 -- less likely, and timing issues with very fast spells (spell
 -- fires before obi is moved) occur at worst on the first spell
 -- not but subsequent ones.
+--
 -- Known bugs: 
--- 1. upon activation, it puts only the first unneeded
--- obi away. The function remove_unneeded_obis() tries to put all
--- of them away, but the calls to itemizer are all made instantly
--- so only the first one is carried out. Usually, only one obi
--- has to be removed per call, so this is not much of a problem.
--- 2. when weather changes due to zoning, get_obis_in_inventory()
--- is called before inventory has loaded and returns nothing. 
+--
+-- 1. Using the get, put, or sort commands too quickly can have
+-- undesirable effects, often leaving obi behind or failing to get
+-- an obi. This is due to limitations on how fast items can be moved
+-- from one inventory to another.
+--
+-- 2. When weather changes due to zoning, get_obi_in_inventory()
+-- is called before inventory has loaded and returns nothing.
 --
 -- 3. Obi is not moved when currently equipped.
--- 
--- To Do: add function to turn the console_echo's on/off. 
+--
+--
+-- To-do:
+--   - Add support for other languages.
+--
+--
 
-_addon.author = "ReaperX"
-_addon.version = "1.0"
-_addon.command = "obiaway"
+-- addon info
+_addon.name = "Obiaway"
+_addon.author = "ReaperX, onetime"
+_addon.version = "1.0.6"
+_addon.commands = {'oa', 'ob', 'obi', 'obiaway'}
+_addon.language = 'english'
 
+-- library includes
 require('sets')
+require('logger')
 config = require('config')
 res = require('resources')
 
-defaults = {}
-defaults.location = 'sack'
+-- settings
+default_settings = {}
+default_settings.notify = true
+default_settings.location = 'sack'
+default_settings.lock = false
+settings = config.load(default_settings)
 
-settings = config.load(defaults)
+-- tokens
+tokens = {}
+tokens.lock_warned = false
+tokens.inv_full_warned = false
+tokens.bag_full_warned = false
 
-windower.register_event('addon command',function (...)
-    if args[1] == 'location' then
-        if S{'sack','case','satchel'}:contains(args[2]:lower()) then
-            settings.location = args[2]
-        else
-            print('Location can only be sack, satchel or case.')
-        end
-    else
-        remove_unneeded_obis()
-    end
-end)
+-- lists
+obi_cities = S{
```

Comment:
If the point of this is to know whether or not certain inventories are available, you can use `windower.ffxi.get_bag_info`. If not, nevermind this comment.